spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java

```diff
@@ -214,9 +248,10 @@ public SvnEnvironmentRepositoryFactory svnEnvironmentRepositoryFactory(Configura
 	}
 
 	@Bean
-	public VaultEnvironmentRepositoryFactory vaultEnvironmentRepositoryFactory(ObjectProvider<HttpServletRequest> request,
-																						EnvironmentWatch watch) {
-		return new VaultEnvironmentRepositoryFactory(request, watch);
+	public VaultEnvironmentRepositoryFactory vaultEnvironmentRepositoryFactory(
```

Comment:
I dont understand the need for this `Bean`, what is using it?

Reply:
@ryanjbaxter With the new list-based composite configuration, we're creating bean definitions dynamically in the `CompositeEnvironmentBeanFactoryPostProcessor`. These factories are used in those bean definitions.

Reply:
OK could we use it above in the `vaultEnvironmentRepository` method as well?  Seems odd that we are recreating the factory there